src/evm/ThresholdVerificationState.sol

```diff
@@ -21,14 +21,15 @@ contract ThresholdVerificationState {
     uint32 guardianSetIndex;
   }
 
-  ThresholdKeyInfo[] private _thresholdKeyData;
+  uint32 private _currentThresholdKeyIndex = type(uint32).max;
```

Comment:
I'd prefer this to be the amount of known keys. I think it's conceptually easier to use that since it's similar to the length of an array.